fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -154,12 +156,20 @@ protected void fillChunkDataBuffer(List<String> patientIds) throws Exception {
                 if (!searchCriteria.isEmpty()) {
                     queryParameters.put(Constants.FHIR_SEARCH_LASTUPDATED, searchCriteria);
                 }
-
                 queryParameters.put("_sort", Arrays.asList(new String[] { Constants.FHIR_SEARCH_LASTUPDATED }));
 
-                for (String patientId : patientIds) {
+                List<String> compartmentSearchCriterias = CompartmentUtil.getCompartmentResourceTypeInclusionCriteria("Patient", resourceType.getSimpleName());
+                if (compartmentSearchCriterias.size() > 1) {
+                    isDoDuplicationCheck = true;
+                }
+
+                for (String compartmentSearchCriteria: compartmentSearchCriterias) {
+                    HashMap<String, List<String>> queryTmpParameters = new HashMap<>();
+                    queryTmpParameters.putAll(queryParameters);
+
+                    queryTmpParameters.put(compartmentSearchCriteria, Arrays.asList(new String[] {String.join(",", patientIds)}));
```

Comment:
took me a second to follow this, but looks OK.

Reply:
:)